packages/engine-core/src/library/LibraryEngine.ts

```diff
@@ -191,6 +197,11 @@ You may have to run ${chalk.greenBright('prisma generate')} for your changes to
         this.QueryEngineConstructor = this.library.QueryEngine
       }
       try {
+        // Using strong reference to `this` inside of log callback will prevent
+        // this instance from being GCed while native engine is alive. At the same time,
+        // `this.engine` field will prevent native instance from being GCed. Using weak ref helps
+        // to avoid this cycle
+        const weakThis = new WeakRef(this)
```

Comment:
Alternative to `WeakRef`: explicitly set `this.engine`  to undefined in `stop` and re-create instance in `start`. This would be more consistent with `BinaryEngine`, where we kill engine process on `stop`, but I was not sure if we concisely diverge here.